curvefs/src/metaserver/inode_storage.h

```diff
@@ -106,12 +111,43 @@ class MemoryInodeStorage : public InodeStorage {
     MetaStatusCode Update(const Inode &inode) override;
     int Count() override;
 
+    bool Save(const std::string& pathname) override;
+
+    bool Load(const std::string& pathname) override;
+
  private:
     RWLock rwLock_;
     // use fsid + inodeid as key
     std::unordered_map<InodeKey, Inode, hashInode> inodeMap_;
 };
 
+class InodeStorageIterator : public Iterator {
```

Comment:
`DentryStorageIterator` and `InodeStorageIterator` are also alomost same except `Hash`, you can extract this to a template class.